app/controllers/api/base_controller/renderer.rb

```diff
@@ -540,6 +546,47 @@ def render_options(resource, data = {})
         klass = collection_class(resource)
         render :json => OptionsSerializer.new(klass, data).serialize
       end
+
+      def virtual_attributes_for(klass)
+        validate_attr_selection(klass).last.reject { |vattr| vattr == "href_slug" }
+      end
+
+      def determine_include_for_find(klass)
+        return nil unless klass.respond_to?(:reflect_on_association)
+
+        type             = @req.subject
+        include_for_find = []
+
+        virtual_attributes_for(klass).each do |vattr|
+          attr_name, attr_base = split_virtual_attribute(vattr)
+          next if attr_base.blank?
```

Comment:
Going to go through each of these individually and explain why they exist:

This one basically is checking if there is a `attr_base`, and if there is, then this is something like `hardware.cpu_sockets`.  In that case, `"hardware"` would be the result here so we would want to keep going.